### PR TITLE
Add some basic devices for i04

### LINKED ISSUE #141: Add i04 devices for knife edge scan
To do knife edge scans on i04 we need at least:
* Smargon
* Gonio positioner
* SDE
* IPIN

### LINKED ISSUE #155: Rename flux PV on i03
As of https://jira.diamond.ac.uk/browse/I03-903 the flux PV on i03 and i04 is the same. We should rename it for main and update https://github.com/DiamondLightSource/dodal/pull/142 to use the central one

### PR DESCRIPTION
**Pull request by DominicOram**
Fixes #141 also fixes #155

To test:
* Run `python tests/system_tests/test_i03_system.py` - confirm devices all connect
* Run `python tests/system_tests/test_i04_system.py` - confirm devices all connect

### CODE REVIEW

File: src/dodal/devices/sample_shutter.py [hunk 1]
```diff
@@ -16,7 +16,9 @@ class SampleShutter(Device):
     pos: EpicsSignal = Component(EpicsSignal, "CTRL2")
     pos_rbv: EpicsSignalRO = Component(EpicsSignalRO, "STA")
 
-    def set(self, open: int):
-        sp_status = self.pos.set(open)
-        rbv_status = await_value(self.pos_rbv, open)
+    def set(self, open_val: int | OpenState):
```
Comment (dperl-dls) [inline src/dodal/devices/sample_shutter.py:19]: can't use this syntax if we wish to support python 3.9

**Review by dperl-dls — COMMENTED** [review]
looks good, just one comment

**Comment by DominicOram** [comment]
Note, many of the comments from @Tom-Willemsen in https://github.com/DiamondLightSource/dodal/pull/160 relate to changes in this PR. I will address them here and rebase/merge in other PR

File: src/dodal/devices/i04/transfocator.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+                # We can only put an integer number of lenses in the beam but the
+                # calculation in the IOC returns the theoretical float number of lenses
+                value = round(value)
+                LOGGER.info(f"Transfocator setting {value} filters")
+                self.number_filters_sp.set(value).wait()
+                self.start.set(1).wait()
+                self.polling_wait_on_start_rbv(1)
+                self.polling_wait_on_start_rbv(0)
+                status.set_finished()
```
Comment (Tom-Willemsen) [inline src/dodal/devices/i04/transfocator.py:71]: See comment in https://github.com/DiamondLightSource/dodal/pull/160#discussion_r1317386357 (sorry I hadn't noticed these PRs were based on each other earlier).

**Review by Tom-Willemsen — APPROVED** [review]
Have somewhat accidentally reviewed this as part of https://github.com/DiamondLightSource/dodal/pull/160, so approved once my one remaining comment here/there is addressed.

**Pull request merged**